pkgs/development/compilers/vala/setup-hook.sh

```diff
@@ -5,8 +5,16 @@ make_vala_find_vapi_files() {
     fi
 }
 
+disable_clang_pointer_warning() {
+    NIX_CFLAGS_COMPILE+=" -Wno-incompatible-function-pointer-types"
```

Comment:
I think it's fine to have this globally for all compilers instead of constraining it to clang.

Reply:
I’ll rework to make it apply unconditionally. It looks like this flag is supported by all clang and GCC versions available in nixpkgs.

Reply:
I pushed an update that adds the flag unconditionally and contains the changes to the setup hook.